okhttp/src/main/java/com/squareup/okhttp/ConnectionSpec.java

```diff
@@ -176,6 +176,7 @@ private ConnectionSpec supportedSpec(SSLSocket sslSocket) {
 
   @Override public boolean equals(Object other) {
     if (!(other instanceof ConnectionSpec)) return false;
+    if (other == this) return true;
```

Comment:
You can actually make this the first statement in the method for an even more optimized optimization!